Merge master into stable-2.x

mprahl opened the pull request.

DharmitD reviewed (commented):
Verified that the package update doesn't affect any other dependencies.

/lgtm

HumairAK commented on go.mod:
```diff
@@ -143,6 +143,7 @@ require (
 )
 
 replace (
+	github.com/go-git/go-git/v5 v5.4.2 => github.com/go-git/go-git/v5 v5.11.0
```
hey @VaniHaripriya question: 

the jira that this links to [here](https://issues.redhat.com/browse/RHOAIENG-6907?jql=project%20%3D%20RHOAIENG%20AND%20component%20%3D%20Pipelines%20AND%20summary%20~%20%22red-hat-data-services%2Fdata-science-pipelines-tekton%22%20AND%20status%20!%3D%20Closed)

identifies this package: github.com/gobuffalo/packr 
which is different, and the suggested fix seems to be on the packr package as well, 

why is go-git updated here instead?

DharmitD replied:
The PR resolves [this jira](https://issues.redhat.com/browse/RHOAIENG-6889)
Points to the go-git package, which is being updated here.

nsingla committed:
Adding K8s Native API Test Plan for ODH

splitting dsp only tests out of kfp pr

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>

adding disconnect RHOAI test section

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>

fixing formatting issue

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>

Renaming test plan file to be ODH specific - to avoid merge conflicts with upstream later on

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>

triggering ci

Signed-off-by: Nelesh Singla <117123879+nsingla@users.noreply.github.com>

openshift-merge-bot[bot] committed:
Merge pull request #195 from nsingla/k8native-dsp

docs(KEP): Splitting Native K8s tests for FIPS and ODH cluster out of KFP PR

mprahl committed:
UPSTREAM <drop>: fix(backend): Fix mutating webhook when pipeline name label value is too long (#12162)

* Fix mutating webhook when pipeline name label value is too long

This just truncates the label value to 63 characters if it's too long.
Since this is a label for convenience and not relied upon in the code
for filtering, this is a safe change.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Free up disk space on upgrade tests to fix CI flakes

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
(cherry picked from commit 7d6722e04b3c95fdc9a7a1250777b404ea2ad80c)

mprahl committed:
UPSTREAM <drop>: fix(backend): Return from pipeline deletions after the cache is updated (#12153)

In Kubernetes native API mode, the REST API could delete a pipeline
version but the cache could not have updated by the time the API call
returns. This adds a 3 second or less polling to ensure the cache gets
updated for a smooth user experience.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
(cherry picked from commit 58709063256c4e9be9f76bce73b4f5964bc19f23)

mprahl committed:
UPSTREAM <drop>: fix(backend/sdk): Fix REST API outputs for pipeline versions with invalid platform specs (#12183)

* Fix REST API outputs for pipeline versions with invalid platform specs

When a platform spec did not contain a `kubernetes` platform when
defined as a Kubernetes manifest (e.g. `spec.platformSpec: {}`),
`YamlStringToPipelineSpecStruct` was errantly detecting it as the
pipeline spec when outputting the pipeline version through the REST API.

This caused the REST API to not return a pipeline spec in the case of it
being empty, and caused the UI to load the pipeline and to properly
create a run.

The solution is when translating from the Kubernetes object to a model
representation, the invalid platform specs are not included. This allows
for proper REST API output while also not modifying the user's spec in
the mutating webhook which would cause GitOps tools to always reconcile.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
(cherry picked from commit 3ea6b60f043efb1050fd882a443f80ffa43bb6dd)

mprahl committed:
UPSTREAM <drop>: fix(backend): Add support for additional filters on the Kubernetes native API (#12189)

* Add support for additional filters on the Kubernetes native API

This adds support for eq, neq, and in filters.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Fix filter mapping in Kubernetes native API

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
(cherry picked from commit 947a988e43ebebc380a06fb179e6b514d87eae5e)

mprahl committed:
Merge pull request #202 from mprahl/cherry-pick-k8s-native

Cherry pick Kubernetes native API fixes

dsp-developers commented:
A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-203`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-203`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-203`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-203`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-203`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-203`

dsp-developers commented:
An OCP cluster where you are logged in as cluster admin is required.

The Data Science Pipelines team recommends testing this using the Data Science Pipelines Operator.  Check [here](https://github.com/opendatahub-io/data-science-pipelines-operator) for more information on using the DSPO.

To use and deploy a DSP stack with these images (assuming the DSPO is deployed), first save the following YAML to a file named `dspa.pr-203.yaml`:
```yaml
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: pr-203
spec:
  dspVersion: v2
  apiServer:
    image: "quay.io/opendatahub/ds-pipelines-api-server:pr-203"
    argoDriverImage: "quay.io/opendatahub/ds-pipelines-driver:pr-203"
    argoLauncherImage: "quay.io/opendatahub/ds-pipelines-launcher:pr-203"
  persistenceAgent:
    image: "quay.io/opendatahub/ds-pipelines-persistenceagent:pr-203"
  scheduledWorkflow:
    image: "quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-203"
  mlmd:  
    deploy: true  # Optional component
    grpc:
      image: "quay.io/opendatahub/mlmd-grpc-server:latest"
    envoy:
      image: "registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2"
  mlpipelineUI:
    deploy: true  # Optional component 
    image: "quay.io/opendatahub/ds-pipelines-frontend:pr-203"
  objectStorage:
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
```

Then run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines.git
cd data-science-pipelines/
git fetch origin pull/203/head
git checkout -b pullrequest 90c5a1cc4cf0f04bdf6ca8274d24ed1ff74918a5
oc apply -f dspa.pr-203.yaml
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.
